Clarify that `if` operates before `matrix` (#27169)
github/docs: data/reusables/actions/jobs/section-using-conditions-to-control-job-execution.md

> Use the `jobs.<job_id>.if` conditional to prevent a job from running unless a condition is met. {% data reusables.actions.if-supported-contexts %}

Reviewer: ```suggestion
You can use the `jobs.<job_id>.if` conditional to prevent a job from running unless a condition is met. {% data reusables.actions.if-supported-contexts %}
```

> You can use the `jobs.<job_id>.if` conditional to prevent a job from running unless a condition is met. {% data reusables.actions.if-supported-contexts %}

Reviewer: ```suggestion
Use the `jobs.<job_id>.if` conditional to prevent a job from running unless a condition is met. {% data reusables.actions.if-supported-contexts %}
```

> {% note %}
> 
> **Note:** This if condition applies before `jobs.<job_id>.matrix`.

Reviewer: ```suggestion
{% note %}

**Note:** The `jobs.<job_id>.if` condition is evaluated before [`jobs.<job_id>.strategy.matrix`](/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix) is applied.
```

You can use the `jobs.<job_id>.if` conditional to prevent a job from running unless a condition is met. {% data reusables.actions.if-supported-contexts %}

> {% note %}
> **Note:** This if condition applies before `jobs.<job_id>.matrix`.
> {% endnote %}

Reviewer: Not rendering correctly in the preview (https://docs-27169-1523dd.preview.ghdocs.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idif). Usually we have blank lines after the opening tag and before the closing tag. I'll see if this fixes it.

```suggestion
{% note %}

**Note:** This if condition applies before `jobs.<job_id>.matrix`.

{% endnote %}
```
